pkg/resources/manifests.go

```diff
@@ -270,6 +270,32 @@ func GenerateDeploymentManifest(ctx context.Context, workspaceObj *kaitov1alpha1
 	labelselector := &v1.LabelSelector{
 		MatchLabels: selector,
 	}
+	initContaiers := []corev1.Container{}
+	envs := []corev1.EnvVar{}
+
+	if len(workspaceObj.Inference.Adapters) != 0 {
+		for i, adapter := range workspaceObj.Inference.Adapters {
+			initContaier := corev1.Container{
+				Name:            fmt.Sprintf("init-container-%d", i+1),
```

Comment:
The init container name should be the adatper.Name.  Hence, in the validation, you need to check to block duplicated names.